--- a/interfaces/opensync.ovsschema
+++ b/interfaces/opensync.ovsschema
@@ -148,6 +148,22 @@
             }
           }
         },
+        "config_apply": {
+          "type": {
+            "key": {
+              "type": "boolean"
+            },
+            "min": 0,
+            "max": 1
+          }
+        },
+        "config_apply_status": {
+          "type": {
+            "key": {
+              "type": "integer"
+            }
+          }
+        },
         "device_mode": {
           "type": {
             "key": {
